c10/xpu/impl/XPUGuardImpl.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <c10/core/DeviceGuard.h>
+#include <c10/core/impl/DeviceGuardImplInterface.h>
+#include <c10/xpu/XPUCachingAllocator.h>
+#include <c10/xpu/XPUFunctions.h>
+#include <c10/xpu/XPUStream.h>
+
+#include <vector>
+
+namespace c10::xpu::impl {
+
+struct XPUGuardImpl final : public c10::impl::DeviceGuardImplInterface {
+  static constexpr DeviceType static_type = DeviceType::XPU;
+
+  XPUGuardImpl() = default;
+
+  explicit XPUGuardImpl(DeviceType t) {
+    TORCH_INTERNAL_ASSERT(t == DeviceType::XPU);
```

Comment:
Note that we have a define from `DeviceType::XPU` to `kXPU`. It is a bit shorter and nicer to read if you want to update this file

Reply:
Thanks, `kXPU` is better. Done.